packages/lib-panoptes-js/src/experimental-auth.js

```diff
@@ -0,0 +1,159 @@
+  const store = _store || globalStore
+  console.log('+++ experimental auth client: _getSession()')
+
+  // TODO
+}
+
+
+/*
+Adds event listener.
+ */
+function addEventListener (eventType, listener, _store) {
+  console.log('+++ experimental auth client: addEventListener()')
+
+  const store = _store || globalStore
+  if (!eventType || !listener) throw new Error('PanoptesJS auth.addEventListener(): requires event type (string) and listener (callback function).')
+
+  // Select array of listeners for specific event type. Create one if it doesn't already exist.
+  if (!store.eventListeners[eventType]) store.eventListeners[eventType] = []
+  const listenersForEventType = store.eventListeners[eventType]
+
+  // Add the callback function to the list of listeners, if it's not already on the list.
+  if (!listenersForEventType.find(l => l === listener)) {
+    listenersForEventType.push(listener)
+  } else {
+    console.log(`PanoptesJS auth.addEventListener(): listener already exists for event type '${eventType}'.`)
+  }
+}
+
+/*
+Remove event listeners.
+ */
+function removeEventListener (eventType, listener, _store) {
+  const store = _store || globalStore
+  console.log('+++ experimental auth client: removeEventListener()')
+
+  // Check if the listener has already been registered for the event type.
+  if (!store.eventListeners[eventType] || !store.eventListeners[eventType]?.find(l => l === listener)) {
+    console.log(`PanoptesJS addEventListener: listener for event type '${eventType}' hasn't been registered.`)
+    return
+  }
+
+  // Remove the listener for that event type.
+  store.eventListeners[eventType] = store.eventListeners[eventType].filter(l => l !== listener)
+  return true
+}
+
+function broadcastEvent (eventType, args, _store) {
+  const store = _store || globalStore
+  store.eventListeners?.[eventType]?.forEach(listener => {
+    listener(args)
+  })
+}
+
+async function signIn (login, password, _store) {
+  // TODO
+  const store = _store || globalStore
+  console.log('+++ experimental auth client: signIn() ', login, password)
+
+  const user = await checkCurrent(_store)
+
+  console.log('+++ user: ', user)
+
+  /*
+  Original PJC code:
+
+  const user = await this.checkCurrent();
+  if (user) {
+    await this.signOut();
+    return this.signIn(credentials);
+  } else {
+    console.log('Signing in', credentials.login);
+    const token = await getCSRFToken(config.host)
+    const data = {
+      authenticity_token: token,
+      user: {
+        login: credentials.login,
+        password: credentials.password,
+        remember_me: true,
+      },
+    };
+
+    const signInRequest = this._makeSignInRequest(data);
+    this._currentUserPromise = signInRequest.catch(() => null);
+    await this._currentUserPromise;
+    this.emit('change', this._currentUserPromise);
+
+    return signInRequest;
+  }
+  */
+}
+
+export {
+  signIn,
+  checkCurrent,
+  addEventListener,
+  removeEventListener,
+}
```

Comment:
If you're going to use ES Module syntax, you'll need to use `type: 'module'` for this package, or rename this file to `experimental-auth.mjs`.

Reply:
Also note that `lib-panoptes-js` is very old, so doesn't use import/export ES syntax anywhere.

Reply:
A big difference between Panoptes-JS and PJC, if you haven't realised this already, is that Panoptes-JS must run in Node and in browsers whereas PJC only runs in browsers.

Panoptes-JS uses `require` and `module.exports` because it was written for older versions of Node. Back in 2018, ESM was supported in browsers but still behind an experimental flag in Node so Panoptes-JS was written in CJS. Since Node 18, you can safely use import/export in Node, if you use `type: 'module'`.